pinot-common/src/main/java/org/apache/pinot/common/function/DateTimePatternHandler.java

```diff
@@ -35,15 +35,33 @@ private DateTimePatternHandler() {
    */
   public static long parseDateTimeStringToEpochMillis(String dateTimeString, String pattern) {
     DateTimeFormatter dateTimeFormatter = getDateTimeFormatter(pattern);
-    return dateTimeFormatter.parseMillis(dateTimeString);
+    return parseMillis(dateTimeString, dateTimeFormatter);
   }
 
   /**
    * Converts the dateTimeString of passed pattern into a long of the millis since epoch
    */
   public static long parseDateTimeStringToEpochMillis(String dateTimeString, String pattern, String timezoneId) {
     DateTimeFormatter dateTimeFormatter = getDateTimeFormatter(pattern, timezoneId);
-    return dateTimeFormatter.parseMillis(dateTimeString);
+    return parseMillis(dateTimeString, dateTimeFormatter);
+  }
+
+  public static long parseDateTimeStringToEpochMillis(String dateTimeString, String pattern, String timezoneId,
+      long defaultVal) {
+    DateTimeFormatter dateTimeFormatter = getDateTimeFormatter(pattern, timezoneId);
+    return parseMillis(dateTimeString, dateTimeFormatter, defaultVal);
+  }
+
+  private static long parseMillis(String dateTimeString, DateTimeFormatter dateTimeFormatter) {
+    return parseMillis(dateTimeString, dateTimeFormatter, 0L);
```

Comment:
We should directly throw exception instead of returning 0. Using implicit default value can case unexpected behavior